// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ==============================================================================

edition = "2023";

package brain.evolution;

import "evolution/lib/individual.proto";

message Nsga2SearchAlgorithmState {
  int64 time_nanos = 1;  // Required.

  // All the incoming individuals before sorting.
  repeated Individual incoming_individuals = 2;

  // All elite individuals.
  repeated Individual outgoing_individuals = 3;
  int64 next_selected_idx = 4;  // Required.

  int64 puts_count = 5;  // Required.
}
